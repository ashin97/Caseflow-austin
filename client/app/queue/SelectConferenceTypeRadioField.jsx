import React, { useState } from 'react';
import PropTypes from 'prop-types';

import RadioField from '../components/RadioField';
import COPY from '../../COPY';

const radioOptions = [
  { displayText: 'Pexip', value: '1' },
  { displayText: 'Webex', value: '2' },
];

const SelectConferenceTypeRadioField = ({ name }) => {
  const [value, setValue] = useState('1');

  return (
    <>
      <RadioField
        label={COPY.USER_MANAGEMENT_SELECT_HEARINGS_CONFERENCE_TYPE}
        name={name}
        options={radioOptions}
        value={value}
        onChange={(newValue) => setValue(newValue)}
        vertical
      />
    </>
  );
};

SelectConferenceTypeRadioField.propTypes = {
  name: PropTypes.string,
};

export default SelectConferenceTypeRadioField;
