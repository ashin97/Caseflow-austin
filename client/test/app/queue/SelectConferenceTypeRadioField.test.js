import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import SelectConferenceTypeRadioField from 'app/queue/SelectConferenceTypeRadioField';

const defaults = {
  name: 'field1',
  value: '1',
  options: [
    { displayText: 'Pexip',
      value: '1' },
    { displayText: 'Webex',
      value: '2' },
  ],
};

describe('SelectConferenceTypeRadioField', () => {
  const handleChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  const setupComponent = (props = {}) => {
    const utils = render(
      <SelectConferenceTypeRadioField
        name={defaults.name}
        onChange={handleChange}
        options={defaults.options}
        {...props}
      />
    );
    const inputs = utils.getAllByRole('radio');

    return {
      inputs,
      ...utils,
    };
  };

  it('renders correctly', async () => {
    const { container } = setupComponent();

    expect(container).toMatchSnapshot();
  });

  it('changes values by radio button selected', async () => {
    setupComponent();
    expect (defaults.value) === "1";

    const webexRadioButton = screen.getByText('Webex');

    await userEvent.click(webexRadioButton);

    expect(defaults.value) === "2";
  })
});
