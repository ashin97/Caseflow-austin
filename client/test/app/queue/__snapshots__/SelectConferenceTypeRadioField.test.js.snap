// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectConferenceTypeRadioField renders correctly 1`] = `
<div>
  <fieldset
    class="usa-fieldset-inputs cf-form-radio "
  >
    <legend
      class=""
    >
      <span>
        Schedule hearings using:
         
      </span>
    </legend>
    <div
      class="cf-form-radio-options"
    >
      <div
        class="cf-form-radio-option"
      >
        <input
          checked=""
          id="field1_1"
          name="field1"
          type="radio"
          value="1"
        />
        <label
          class=""
          for="field1_1"
        >
          Pexip
        </label>
      </div>
      <div
        class="cf-form-radio-option"
      >
        <input
          id="field1_2"
          name="field1"
          type="radio"
          value="2"
        />
        <label
          class=""
          for="field1_2"
        >
          Webex
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;
